tools/runners/test262-harness.py

```diff
@@ -613,7 +620,8 @@ def run_test_in(self, tmp):
         command = TestCase.instantiate_template(self.command_template, {
             'path': tmp.name
         })
-        (code, out, err) = TestCase.execute(command)
+        # Only es2015 or esnext need the timeout option.
+        (code, out, err) = TestCase.execute(command, 60)
```

Comment:
Add timeout option for the usage in es5.1,
setting to 60 for pass two time consuming tests.

Reply:
I'm not sure if it is a good idea to have 60 seconds timeout unconditionally,
because language/statements/for-of/iterator-next-reference.js in ES2015-test262
is a wrong test (outdated because of spec change) and stucks in an infinite loop.

Increasing timeout means that running ES2015-test262 will take an extra minute because of this wrong test.
8 seconds timeout is enough to remove 2 slow tests from es2015 excludelist. And tests run in 80 seconds
for me with 8 seconds timeout, but in 80+57 seconds with 60 seconds timeout.

In nutshell: I would keep general timeout as low as possible, but we could add extra timeout for ES5.1 if it is necessary.

Reply:
60 is needed to pass two new tests

Reply:
> I'm not sure if it is a good idea to have 60 seconds timeout unconditionally,
> because language/statements/for-of/iterator-next-reference.js in ES2015-test262
> is a wrong test (outdated because of spec change) and stucks in an infinite loop.
> 
> Increasing timeout means that running ES2015-test262 will take an extra minute because of this wrong test.
> 8 seconds timeout is enough to remove 2 slow tests from es2015 excludelist. And tests run in 80 seconds
> for me with 8 seconds timeout, but in 80+57 seconds with 60 seconds timeout.
> 
> In nutshell: I would keep general timeout as low as possible, but we could add extra timeout for ES5.1 if it is necessary.

60s needed to pass
  <test id="built-ins/decodeURIComponent/S15.1.3.2_A2.5_T1.js"><reason></reason></test>	
  <test id="built-ins/decodeURI/S15.1.3.1_A2.5_T1.js"><reason></reason></test>

These two tests consume lot of time in jerryscript, maybe it's a performance issue.

Reply:
The bottleneck is the wrong language/statements/for-of/iterator-next-reference.js in test262-es2015. If we increase the timeout, we unnecessarily increase the runtime, because we have to wait to kill this wrong test. Otherwise the 2 unskipped es2015 test is also covered by esnext, because test262 maintainers know these are slow tests and refactored them to be faster. So we don't loose anything if we keep the timeout as is and leave these tests skipped.

Reply:
> The bottleneck is the wrong language/statements/for-of/iterator-next-reference.js in test262-es2015. If we increase the timeout, we unnecessarily increase the runtime, because we have to wait to kill this wrong test. Otherwise the 2 unskipped es2015 test is also covered by esnext, because test262 maintainers know these are slow tests and refactored them to be faster. So we don't loose anything if we keep the timeout as is and leave these tests skipped.

Not that simple, the debug version build jerryscript are slower, we need more time tolerance, and that won't increate the total test262 running time much, because tests are running in parallel, the testes consume lot of time won't be a bottleneck.

Reply:
@ossy-szeged Revert to 5 secons.